app/components/autocomplete/at_mention_item/at_mention_item.js

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    Text,
+    TouchableOpacity,
+    View
+} from 'react-native';
+
+import ProfilePicture from 'app/components/profile_picture';
+import {makeStyleSheetFromTheme, changeOpacity} from 'app/utils/theme';
+
+export default class AtMentionItem extends PureComponent {
+    static propTypes = {
+        firstName: PropTypes.string,
+        lastName: PropTypes.string,
+        onPress: PropTypes.func.isRequired,
+        userId: PropTypes.string.isRequired,
+        username: PropTypes.string,
+        theme: PropTypes.object.isRequired
+    };
+
+    completeMention = (username) => {
+        this.props.onPress(username);
+    };
+
+    render() {
+        const {
+            firstName,
+            lastName,
+            userId,
+            username,
+            theme
+        } = this.props;
+
+        const style = getStyleFromTheme(theme);
+        const hasFullName = firstName.length > 0 && lastName.length > 0;
+
+        return (
+            <TouchableOpacity
+                key={userId}
+                onPress={this.completeMention.bind(this, username)}
```

Comment:
Why are we passing username in? `this.completeMention` can access the props itself